packages/eslint-plugin/src/rules/keyword-spacing.ts

```diff
@@ -31,7 +31,10 @@ export default util.createRule<Options, MessageIds>({
     const baseRules = baseRule.create(context);
     return {
       ...baseRules,
-      TSAsExpression(node): void {
+      TSTypeAssertion(node): void {
+        if (node.kind !== 'as') {
+          return;
+        }
```

Comment:
could do this via a selector

```suggestion
      'TSTypeAssertion[kind = "as"]'(node): void {
```

Reply:
we could also add this as a typed selector to `packages/types/src/ast-node-types.ts` as it's probably going to be a more common usecase